plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts

```diff
@@ -0,0 +1,283 @@
+
+	document
+		.querySelector( '#post-preview' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_preview_changes' );
+		} );
+
+	document
+		.querySelector( '.submitduplicate' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_copy', getProductData() );
+		} );
+
+	document
+		.querySelector( '.submitdelete' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_delete', getProductData() );
+		} );
+
+	document
+		.querySelectorAll(
+			'.edit-post-status, .edit-visibility, .edit-timestamp, .edit-catalog-visibility'
+		)
+		.forEach( ( button ) => {
+			button.addEventListener( 'click', () => {
+				recordEvent( 'product_publish_widget_edit', {
+					...getPublishingWidgetData(),
+					...getProductData(),
+				} );
+			} );
+		} );
+
+	document.querySelector( '#publish' )?.addEventListener( 'click', () => {
+		recordEvent( 'product_publish_widget_save', {
+			...prefixObjectKeys( getPublishingWidgetData(), 'new_' ),
+			...prefixObjectKeys( initialPublishingData, 'current_' ),
+			...getProductData(),
+		} );
+	} );
+
+	document
+		.querySelectorAll( '.handle-order-lower, .handle-order-higher' )
+		.forEach( ( button ) => {
+			button.addEventListener( 'click', ( event ) => {
+				const postBox = ( event.target as HTMLElement ).closest(
+					'.postbox'
+				);
+
+				if ( ! postBox ) {
+					return;
+				}
+
+				recordEvent( 'product_widget_order_change', {
+					widget: postBox.id,
+				} );
+			} );
+		} );
+
+	document
+		.querySelector( '#show-settings-link' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_screen_options_open' );
+		} );
+
+	document
+		.querySelectorAll( '#adv-settings .metabox-prefs input[type=checkbox]' )
+		.forEach( ( input ) => {
+			input.addEventListener( 'change', () => {
+				recordEvent( 'product_screen_elements', {
+					selected_element: ( input as HTMLInputElement ).value,
+					checkbox: ( input as HTMLInputElement ).checked,
+				} );
+			} );
+		} );
+
+	document
+		.querySelectorAll( 'input[name="screen_columns"]' )
+		.forEach( ( input ) => {
+			input.addEventListener( 'change', () => {
+				recordEvent( 'product_layout', {
+					selected_layout: ( input as HTMLInputElement ).value,
+				} );
+			} );
+		} );
+
+	document
+		.querySelector( '#editor-expand-toggle' )
+		?.addEventListener( 'change', ( event ) => {
+			recordEvent( 'product_additional_settings', {
+				checkbox: ( event.target as HTMLInputElement ).checked,
+			} );
+		} );
+
+	// Product tags
+
+	document
+		.querySelector( '.tagadd' )
+		?.addEventListener( 'click', ( event ) => {
+			const tagInput = document.querySelector< HTMLInputElement >(
+				'#new-tag-product_tag'
+			);
+			if ( tagInput && tagInput.value && tagInput.value.length > 0 ) {
+				recordEvent( 'product_tags_add', {
+					page: 'product',
+					tag_string_length: tagInput.value.length,
+					tag_list_size:
+						( document.querySelector( '.tagchecklist' )?.children
+							.length || 0 ) + 1,
```

Comment:
I'm not sure about the + 1. After testing this on my end, `tag_list_size` is sending an extra item than it should.

Reply:
Fixed as part of this commit: https://github.com/woocommerce/woocommerce/pull/33121/commits/8f61bef2e0e8b947dcc3b6cfccb49c160647ffdc